modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/DefaultSourceCallbackContext.java

```diff
@@ -165,9 +166,26 @@ public void setCorrelationId(String correlationId) {
     return sourceCallback;
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void fire(NotificationActionDefinition<?> action, TypedValue<?> data) {
+    sourceNotification = new SourceNotification(action, data);
```

Comment:
this doesn't seem right. If i call this method twice, i expect both notifications to be fired, and I expect them to be fired at the moment i call this method. Why can't you just fire it?